packages/react-composites/src/composites/ChatComposite/ChatScreen.tsx

```diff
@@ -202,28 +205,37 @@ export const ChatScreen = (props: ChatScreenProps): JSX.Element => {
   const headerProps = useAdaptedSelector(getHeaderProps);
   const errorBarProps = usePropsFor(ErrorBar);
 
+  const overlayImageItemRef = React.useRef<OverlayImageItem | undefined>(overlayImageItem);
+  overlayImageItemRef.current = overlayImageItem;
```

Comment:
Do we need this line?
`overlayImageItemRef.current = overlayImageItem;`
it looks like the value is updated below

Reply:
Before it was used in the useEffect, but then required the: 
> // Disable eslint because we are using the overlayImageItem in this effect but don't want to have it as a dependency, as it will cause an infinite loop.

Storing as a ref avoids this problem, however any render pass we do want it to use any latest value so holding in a ref circumvents the useEffect dep but keeps the value up to date

> it looks like the value is updated below
Should only be updated in these two lines. The first line technically don't need to pass it into the ref contructor, but this contructor only applies when the component first mounts. Then this line (`overlayImageItemRef.current = overlayImageItem`) ensures the value is kept up to date when it needs to be

Reply:
I think we don't want to have the latest value here until the overlay was closed and reopened (at least it looks like this from the code and what I remember from bug bashes)
Adding @Leah-Xia-Microsoft here to clarify the expected behaviour

Love the useRef change though!

Reply:
> I think we don't want to have the latest value here until the overlay was closed and reopened (at least it looks like this from the code and what I remember from bug bashes) Adding @Leah-Xia-Microsoft here to clarify the expected behaviour
> 
> Love the useRef change though!

That's correct. We don't need the latest value for overlay. Teams behaves the same way. For example, if a user opens the overlay with image A, and this image got deleted from some other client while the overlay is open, user A should still see image A in the overlay until he/she close the overlay.

Reply:
Hmm, so when do you want `overlayImageItem` to update?

Before this change it was updated any time `messageThreadProps.messages` changed but the comment referred to this being left out of the dependencies purely to avoid an infinte update loop (so switched to using ref to avoid)

Reply:
Note the ref tracks the latest value, but this useEffect is still only kicked off at the same time (when state.thread.messages updates) so there shouldn't be any functional difference with this change

Reply:
> Hmm, so when do you want `overlayImageItem` to update?
> 
> Before this change it was updated any time `messageThreadProps.messages` changed but the comment referred to this being left out of the dependencies purely to avoid an infinte update loop (so switched to using ref to avoid)

@JamesBurnside, looked into your changes, I agree these changes will not change the current behaviour. I've also build and tested on your branch. Everything works as expected. Thanks for the fix.

@vhuseinova-msft 
`overlayImageItem` is only updated when the fullSizeImageSrc is ready.
When `onInlineImageClicked` is triggered, we want to open the overlay right away for a better UI experience. At that moment, the fullSizeImageSrc might not be ready.
Only when `messageThreadProps.messages` is changed **AND** when the following statement is true
`if (overlayImageItem.imageSrc === '' && resourceCache && resourceCache[overlayImageItem.imageUrl])`,
we update the `overlayImageItem`.